public/browser-app.js

```diff
@@ -0,0 +1,87 @@
+const tasksDOM = document.querySelector('.tasks')
+const loadingDOM = document.querySelector('.loading-text')
+const formDOM = document.querySelector('.task-form')
+const taskInputDOM = document.querySelector('.task-input')
+const formAlertDOM = document.querySelector('.form-alert')
+// Load tasks from /api/tasks
+const showTasks = async () => {
+  loadingDOM.style.visibility = 'visible'
+  try {
+    const {
+      data: { tasks },
+    } = await axios.get('/api/v1/tasks')
+    if (tasks.length < 1) {
+      tasksDOM.innerHTML = '<h5 class="empty-list">No tasks in your list</h5>'
+      loadingDOM.style.visibility = 'hidden'
+      return
+    }
+    const allTasks = tasks
+      .map((task) => {
+        const { completed, _id: taskID, name } = task
+        return `<div class="single-task ${completed && 'task-completed'}">
+<h5><span><i class="far fa-check-circle"></i></span>${name}</h5>
+<div class="task-links">
+
+
+
+<!-- edit link -->
+<a href="task.html?id=${taskID}"  class="edit-link">
+<i class="fas fa-edit"></i>
+</a>
+<!-- delete btn -->
+<button type="button" class="delete-btn" data-id="${taskID}">
+<i class="fas fa-trash"></i>
+</button>
+</div>
+</div>`
+      })
+      .join('')
+    tasksDOM.innerHTML = allTasks
+  } catch (error) {
+    tasksDOM.innerHTML =
+      '<h5 class="empty-list">There was an error, please try later....</h5>'
+  }
+  loadingDOM.style.visibility = 'hidden'
+}
+
+showTasks()
+
+// delete task /api/tasks/:id
+
+tasksDOM.addEventListener('click', async (e) => {
+  const el = e.target
+  if (el.parentElement.classList.contains('delete-btn')) {
+    loadingDOM.style.visibility = 'visible'
+    const id = el.parentElement.dataset.id
+    try {
+      await axios.delete(`/api/v1/tasks/${id}`)
+      showTasks()
+    } catch (error) {
+      console.log(error)
```

Comment:
Don't use console.log or alerts in your frontend code. This is only for debbuging. For a real user, if some error happend, you want to send the data to the server, print something to the user, or doing nothing. logging it to the cosole or alerting it will never help, and will show you application as clumsy for sophisticated users.

Reply:
באופן כללי, אני חושב שהסיבה לכך שהפרונט מבולגן הוא שהמרצה ניסה לעשות פרונט בלי להשתמש בדברים שאנחנו כביכול עוד לא יודעים עכשיו. הוא מסתמך על כך שאנחנו יודעים JS ולכן אמורים להכיר רק דברים שנעשים ב DOM ו QUERYSELECTOR וכאלה. אני מאמין שבקורס של REACT הוא כבר יהיה הרבה יותר מסודר